plugins/catalog-backend/migrations/20200511113813_init.js

```diff
@@ -103,6 +103,9 @@ exports.up = async function up(knex) {
         table.comment(
           'Flattened key-values from the entities, used for quick filtering',
         );
+        table
+          .increments('id', { primaryKey: true })
+          .comment('Primary key to distinguish unique lines from each other');
```

Comment:
Out of curiosity, what is your objection to adding an `id` column? Because it isn't used anywhere? Or because it isn't needed if we add a composite primary key?

FWIW, I like the idea of using a composite primary key, I'm mostly just curious about the guiding philosophy, so we can apply it to the rest of the PR. Thanks!

Reply:
@freben @Gerg It looks like this table `entities_search` and the `entity` table are both permanently dropped (along with `entities_relations`) in a more recent [migration](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/migrations/20220116144621_remove_legacy.js#L24). 

Does the implementation matter given that it seem that it will make no difference to end state of the schemas?